app/services/follow_remote_account_service.rb

```diff
@@ -28,21 +28,30 @@ def call(uri)
 
     return Account.find_local(confirmed_username) if TagManager.instance.local_domain?(confirmed_domain)
 
-    confirmed_account = Account.find_remote(confirmed_username, confirmed_domain)
-    return confirmed_account unless confirmed_account.nil?
+    remote_url  = data.link('http://schemas.google.com/g/2010#updates-from').href
+    salmon_url  = data.link('salmon').href
+    url         = data.link('http://webfinger.net/rel/profile-page').href
+    public_key  = magic_key_to_pem(data.link('magic-public-key').href)
 
-    Rails.logger.debug "Creating new remote account for #{uri}"
+    account = Account.find_remote(confirmed_username, confirmed_domain)
+    return account unless account.nil? || remote_url != account.remote_url || salmon_url != account.salmon_url || url != account.url || public_key != account.public_key
```

Comment:
You're past the webfinger fetch here, at least (`Goldfinger.finger("acct:#{uri}")`).

The feed fetch is `get_feed(account.remote_url)`, perhaps you could make it conditional on `remote_url` having changed.

Generally, yes, I am concerned about this service's performance, because it is called quite a lot, and we want to minimize extra network requests. Perhaps some of this work could be off-loaded into a background worker, but it wouldn't do to queue up thousands of duplicates, so there should be some sort of check on that. Perhaps something like `last_webfingered_at` and doing it only once a day or week or month?

Just throwing ideas out there, this is not a definitive suggestion.